src/main/java/org/jpeek/XslReport.java

```diff
@@ -0,0 +1,249 @@
+import com.jcabi.xml.XSLChain;
+import com.jcabi.xml.XSLDocument;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import org.cactoos.collection.CollectionOf;
+import org.cactoos.io.ResourceOf;
+import org.cactoos.io.TeeInput;
+import org.cactoos.scalar.LengthOf;
+import org.cactoos.text.FormattedText;
+import org.cactoos.text.TextOf;
+import org.xembly.Directives;
+import org.xembly.Xembler;
+
+/**
+ * Single report.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Yegor Bugayenko (yegor256@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle JavadocTagsCheck (500 lines)
+ */
+final class XslReport implements Report {
+    /**
+     * Default mean.
+     */
+    private static final double DEFAULT_MEAN = 0.5d;
+
+    /**
+     * Default sigma.
+     */
+    private static final double DEFAULT_SIGMA = 0.1d;
+
+    /**
+     * Location to the schema file.
+     */
+    private static final String SCHEMA_FILE = "xsd/metric.xsd";
+
+    /**
+     * XSD schema.
+     */
+    private static final XSD SCHEMA = XSDDocument.make(
+        XslReport.class.getResourceAsStream(XslReport.SCHEMA_FILE)
+    );
+
+    /**
+     * XSL stylesheet.
+     */
+    private static final XSL STYLESHEET = XSLDocument.make(
+        XslReport.class.getResourceAsStream("xsl/metric.xsl")
+    ).with(new ClasspathSources());
+
+    /**
+     * The skeleton.
+     */
+    private final XML skeleton;
+
+    /**
+     * The metric.
+     */
+    private final String metric;
+
+    /**
+     * Post processing XSLs.
+     */
+    private final XSL post;
+
+    /**
+     * XSL params.
+     */
+    private final Map<String, Object> params;
+
+    /**
+     * Ctor.
+     * @param xml Skeleton
+     * @param name Name of the metric
+     */
+    XslReport(final XML xml, final String name) {
+        this(
+            xml, name, new HashMap<>(0),
+            XslReport.DEFAULT_MEAN, XslReport.DEFAULT_SIGMA
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param xml Skeleton
+     * @param name Name of metric
+     * @param args Params for XSL
+     */
+    XslReport(final XML xml, final String name, final Map<String, Object> args) {
+        this(
+            xml, name, args,
+            XslReport.DEFAULT_MEAN, XslReport.DEFAULT_SIGMA
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param xml Skeleton
+     * @param name Name of the metric
+     * @param args Params for XSL
+     * @param mean Mean
+     * @param sigma Sigma
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    XslReport(final XML xml, final String name,
+        final Map<String, Object> args,
+        final double mean, final double sigma) {
+        this.skeleton = xml;
+        this.metric = name;
+        this.params = args;
+        this.post = new XSLChain(
+            new CollectionOf<>(
+                new XSLDocument(
```

Comment:
Use new XSLDocument(new TextOf(new ResourceOf(...))) ....similar to method xml() at line 234 down below so that it looks uniform across the file.